addons/hr_holidays/tests/test_working_hours.py

```diff
@@ -78,14 +78,15 @@ def test_multi_companies_2_employees_2_selected_companies_company_holidays(self)
         self.env.user.company_id = self.company_A
         self.env.user.company_ids = [self.company_A.id, self.company_B.id]
 
-        self.env['hr.leave'].create({
+        company_leave = self.env['hr.leave.generate.multi.wizard'].create({
             'name': 'holiday from monday to tuesday',
-            'holiday_type': 'company',
-            'mode_company_id': self.company_A.id,
+            'allocation_mode': 'company',
+            'company_id': self.company_A.id,
             'holiday_status_id': self.leave_type.id,
-            'request_date_from': datetime(2023, 12, 25),
-            'request_date_to': datetime(2023, 12, 26, 23, 59, 59),
+            'date_from': datetime(2023, 12, 25),
+            'date_to': datetime(2023, 12, 26, 23, 59, 59),
```

Comment:
As the field is storing dates, you should probably use dates here:
```suggestion
            'date_from': date(2023, 12, 25),
            'date_to': date(2023, 12, 26),
```